File: bot/src/main/java/com/ryderbelserion/discord/api/embeds/EmbedColor.java
package com.ryderbelserion.discord.api.embeds;

import com.ryderbelserion.discord.api.utils.ColorUtils;
import org.jspecify.annotations.NonNull;
import java.awt.*;

public enum EmbedColor {

    DEFAULT("#bff7fd"),
    SUCCESS("#0eeb6a"),
    FAIL("#e0240b"),
    WARNING("#eb6123"),
    EDIT("#5e68ff");

    private final Color color;

    EmbedColor(@NonNull final String code) {
        this.color = ColorUtils.toColor(code);
    }

    public @NonNull final Color getColor() {
        return this.color;
    }
}

File: src/main/kotlin/com/ryderbelserion/api/builders/embeds/EmbedColor.kt
package com.ryderbelserion.api.builders.embeds

import com.ryderbelserion.api.util.ColorUtil.toColor
import java.awt.Color

enum class EmbedColor(code: String) {
    DEFAULT("#bff7fd"),
    SUCCESS("#0eeb6a"),
    FAIL("#e0240b"),
    WARNING("#eb6123"),
    EDIT("#5e68ff");

    val color: Color = code.toColor()
}